addons/binding/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/handler/KNXGenericThingHandler.java

```diff
@@ -0,0 +1,1124 @@
+
+                Thread.sleep(OPERATION_INTERVAL);
+                byte[] objecttableaddress = ((KNXBridgeBaseThingHandler) getBridge().getHandler()).readDeviceProperties(
+                        address, GROUPOBJECT_OBJECT, PID.TABLE_REFERENCE, 1, 1, true, OPERATION_TIMEOUT);
+
+                if (objecttableaddress != null) {
+                    Thread.sleep(OPERATION_INTERVAL);
+                    elements = ((KNXBridgeBaseThingHandler) getBridge().getHandler()).readDeviceMemory(address,
+                            toUnsigned(objecttableaddress), 1, false, OPERATION_TIMEOUT);
+                    if (elements != null) {
+                        int numberOfElements = toUnsigned(elements);
+                        logger.debug("The KNX Actor with address {} has {} objects", address, numberOfElements);
+
+                        for (int i = 1; i < numberOfElements; i++) {
+                            byte[] objectData = null;
+                            while (objectData == null) {
+                                Thread.sleep(OPERATION_INTERVAL);
+                                objectData = ((KNXBridgeBaseThingHandler) getBridge().getHandler()).readDeviceMemory(
+                                        address, toUnsigned(objecttableaddress) + 1 + (i * 3), 3, false,
+                                        OPERATION_TIMEOUT);
+
+                                logger.debug("Byte 1 {}", String
+                                        .format("%8s", Integer.toBinaryString(objectData[0] & 0xFF)).replace(' ', '0'));
+                                logger.debug("Byte 2 {}", String
+                                        .format("%8s", Integer.toBinaryString(objectData[1] & 0xFF)).replace(' ', '0'));
+                                logger.debug("Byte 3 {}", String
+                                        .format("%8s", Integer.toBinaryString(objectData[2] & 0xFF)).replace(' ', '0'));
+                            }
+                        }
+                    }
+                } else {
+                    logger.warn("The KNX Actor with address {} does not expose a Group Object table", address);
+                }
+
+                filledDescription = true;
+
+            } catch (Exception e) {
+                logger.debug("An exception occurred while fetching the device description for a Thing '{}' : {}",
+                        getThing().getUID(), e);
+                e.printStackTrace();
+            }
+        }
+
+        private int toUnsigned(final byte[] data) {
+            int value = data[0] & 0xff;
+            if (data.length == 1) {
+                return value;
+            }
+            value = value << 8 | data[1] & 0xff;
+            if (data.length == 2) {
+                return value;
+            }
+            value = value << 16 | data[2] & 0xff << 8 | data[3] & 0xff;
+            return value;
+        }
+    };
+
+    public enum LoadState {
+        L0(Integer.valueOf(0), "Unloaded"),
```

Comment:
I think it was due to me not wanting to use EnumMap or depend on the order of the Enums wrt .ordinal()